### PR TITLE
fix(restapi) add archived case api

### PR DESCRIPTION
**Pull request by cpuy**
Add documentation for archived case context.

Closes [BS-15979](https://bonitasoft.atlassian.net/browse/BS-15979)

### CODE REVIEW

**Comment by laurentleseigneur** [comment]
retest this please

File: md/bpm-api.md [hunk 1]
```diff
@@ -3092,6 +3092,44 @@ You can search cases.
...
+    ```json
+    {  
+      "businessData_ref":{  
+        "name":"myBusinessDate",
+        "type":"com.company.model.BusinessObject1",
+        "link":"API/bdm/businessData/com.company.model.BusinessObject1/2",
+        "storageId":2,
+        "storageId_string":"2"
+      },
+      "document1_ref":{
```
Comment (laurentleseigneur) [inline md/bpm-api.md]: this part seems to be the response and not the payload ?

Comment (cpuy) [reply md/bpm-api.md]: Actually this is the response payload :-).  So i guess using payload here is right

Comment (laurentleseigneur) [reply md/bpm-api.md]: your example is false: is name is `myBusinessDate`, reference will be `myBusinessDate_ref` in json response

Comment (laurentleseigneur) [reply md/bpm-api.md]: see real example here:
> {
> "ss_ref": {
> "name": "ss",
> "type": "com.company.model.BusinessObject1",
> "link": "API/bdm/businessData/com.company.model.BusinessObject1/2",
> "storageId": 2,
> "storageId_string": "2"
> },
> "mydoc_ref": {
> "id": 101,
> "processInstanceId": 1002,
> "name": "mydoc",
> "author": 4,
> "creationDate": 1493717008671,
> "fileName": "doc.pdf",
> "contentMimeType": "application/octet-stream",
> "contentStorageId": "101",
> "url": "documentDownload?fileName=doc.pdf&contentStorageId=101",
> "description": "dwcxdwcxdv",
> "version": "1",
> "index": -1,
> "contentFileName": "doc.pdf"
> }
> }

**Review by laurentleseigneur — CHANGES_REQUESTED** [review]

File: md/bpm-api.md (hunk 1 above)
Comment (cpuy) [reply md/bpm-api.md]: That is true.
Wrong copy paste from case context.

I'll fix this section as well

**Comment by cpuy** [comment]
See https://github.com/bonitasoft/bonita-doc/pull/278 to fix typos in context APIs

**Comment by cpuy** [comment]
UP!

**Review by laurentleseigneur — APPROVED** [review]

**Comment by laurentleseigneur** [comment]
don't forget to merge doc branches (->7.4,7.5,7.6)

**Pull request merged**